warehouse/packaging/models.py

```diff
@@ -194,6 +197,35 @@ def documentation_url(self):
 
         return request.route_url("legacy.docs", project=self.name)
 
+    @property
+    def all_releases(self):
+        return sorted(
+            orm.object_session(self).query(Release)
+            .filter(Release.project == self)
+            .options(orm.load_only(
+                Release.version,
+                Release.is_prerelease,
+                Release._pypi_hidden,
+                Release._pypi_ordering,
+                Release.created))
+            .order_by(Release._pypi_ordering.desc())
+            .all(),
+            key=lambda r: parse_version(r.version),
+            reverse=True,
+        )
+
+    @property
+    def latest_release(self):
+        return first(
+            sorted(
+                self.all_releases,
```

Comment:
I believe that this is going to cause things that already use ``Project.all_releases`` to re-execute the ``all_releases`` query instead of re-using the already fetched values like it did previously. If we're going to end up executing a second query, we might as well handle all of the sorting in SQL. I also ask a similar question, is the only thing we *really* care about getting the latest version? then maybe this would be better off specified as:


```python
@property
def latest_version(self):
    return (orm.object_session(self)
               .query(Release.version)
               .filter(Release.project = self)
               .order_by(
                   Release.is_prerelease.nullslast(),
                   Release._pypi_ordering.desc())
               .first())
```